desktop/app/store-helper.js

```diff
@@ -1,10 +1,37 @@
 import {ipcMain} from 'electron'
 
-export default function () {
-  // In case the subscribe store comes before the remote store is ready
+export default function (mainWindow) {
+  const subscribeStoreSubscribers = []
+  let store = {}
+
   ipcMain.on('subscribeStore', event => {
```

Comment:
We now move the job of updating the other browserwindows's store to the main thread. We still keep this all in the main renderer but its not sending it directly to the other windows (which turns out to be sync due to the remote proxy stuff)